File: dev/sbt-scripted-library/src/main/java/impl/FooServiceImpl.java
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package impl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import java.util.concurrent.CompletableFuture;
import api.FooService;

public class FooServiceImpl implements FooService {

  @Override
  public ServiceCall<NotUsed, NotUsed> foo() {
    return request -> CompletableFuture.completedFuture(NotUsed.getInstance());
  }
}

File: dev/sbt-plugin/src/sbt-test/sbt-plugin/service-discovery/a/impl/src/main/scala/impl/FooServiceImpl.scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package impl

import com.lightbend.lagom.scaladsl.api.ServiceCall
import api.FooService
import scala.concurrent.duration._
import akka.discovery.Discovery
import akka.actor.ActorSystem

class FooServiceImpl(actorSystem: ActorSystem) extends FooService {

  implicit val ec = actorSystem.dispatcher

  override def foo = ServiceCall { _ =>
    val serviceDiscovery = Discovery(actorSystem).discovery
    serviceDiscovery.lookup("foo-service", 100.milliseconds).map { res =>
      res.serviceName
    }
  }
}
